llvm/lib/Transforms/Scalar/InferAddressSpaces.cpp

```diff
@@ -145,6 +145,11 @@ static cl::opt<bool> AssumeDefaultIsFlatAddressSpace(
 static const unsigned UninitializedAddressSpace =
     std::numeric_limits<unsigned>::max();
 
+static cl::opt<int>
+    AddressSpaceArg("address-space", cl::init(UninitializedAddressSpace),
```

Comment:
I can't see this option is used even in the test you added. Should we remove this change leaving only the fix and the test?

Reply:
fixed that